lib/components/component.dart

```diff
@@ -23,7 +23,7 @@ abstract class Component {
   @mustCallSuper
   void update(double dt) {
     _effects.removeWhere((e) => e.hasFinished());
-    _effects.forEach((e) => e.update(dt));
+    _effects.where((e) => !e.isPaused).forEach((e) => e.update(dt));
```

Comment:
Minor suggestion, to avoid multiple iterations (a where and then a forEach). We could check on the update if the effect is paused. What do you think?

Reply:
Isn't the Dart compiler smart enough to bundle them? Before I had it like this:
`effect.forEach((e) => if(!e.isPaused) e.update(dt));`
but I thought the current solution was easier to read?

Reply:
I don't know if the dart compiler will be smart or not for this, but I would bet that it don't on this case.

Anyway, I liked the new approach 🎉

Reply:
we had that discussion before, if dart optimizes that like java streams. my conclusion is: unless it's a core loop, let's go for readability. this is one components effects. it will likely never matter.

if there is something we can optimize in the for each component in the game loop, thats where we should invest